matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/location/RedactLiveLocationShareTask.kt

```diff
@@ -0,0 +1,78 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.android.sdk.internal.session.room.location
+
+import io.realm.RealmConfiguration
+import org.matrix.android.sdk.internal.database.awaitTransaction
+import org.matrix.android.sdk.internal.database.model.livelocation.LiveLocationShareAggregatedSummaryEntity
+import org.matrix.android.sdk.internal.database.query.get
+import org.matrix.android.sdk.internal.di.SessionDatabase
+import org.matrix.android.sdk.internal.session.room.send.LocalEchoEventFactory
+import org.matrix.android.sdk.internal.session.room.send.queue.EventSenderProcessor
+import org.matrix.android.sdk.internal.task.Task
+import timber.log.Timber
+import javax.inject.Inject
+
+internal interface RedactLiveLocationShareTask : Task<RedactLiveLocationShareTask.Params, Unit> {
+    data class Params(
+            val roomId: String,
+            val beaconInfoEventId: String,
+            val reason: String?
+    )
+}
+
+internal class DefaultRedactLiveLocationShareTask @Inject constructor(
+        @SessionDatabase private val realmConfiguration: RealmConfiguration,
+        private val localEchoEventFactory: LocalEchoEventFactory,
+        private val eventSenderProcessor: EventSenderProcessor,
+) : RedactLiveLocationShareTask {
+
+    override suspend fun execute(params: RedactLiveLocationShareTask.Params) {
+        val relatedEventIds = getRelatedEventIdsOfLive(params.beaconInfoEventId)
+        Timber.d("beacon with id ${params.beaconInfoEventId} has related event ids: ${relatedEventIds.joinToString(", ")}")
+
+        redactEvent(
+                eventId = params.beaconInfoEventId,
+                roomId = params.roomId,
+                reason = params.reason
+        )
+        relatedEventIds.forEach { eventId ->
+            redactEvent(
+                    eventId = eventId,
+                    roomId = params.roomId,
+                    reason = params.reason
+            )
+        }
+    }
+
+    private suspend fun getRelatedEventIdsOfLive(beaconInfoEventId: String): List<String> {
+        return awaitTransaction(realmConfiguration) { realm ->
+            val aggregatedSummaryEntity = LiveLocationShareAggregatedSummaryEntity.get(
+                    realm = realm,
+                    eventId = beaconInfoEventId
+            )
+            aggregatedSummaryEntity?.relatedEventIds?.toList() ?: emptyList()
+        }
+    }
+
+    private fun redactEvent(eventId: String, roomId: String, reason: String?) {
+        Timber.d("redacting event of id $eventId")
+        val redactionEcho = localEchoEventFactory.createRedactEvent(roomId, eventId, reason)
+        localEchoEventFactory.createLocalEcho(redactionEcho)
```

Comment:
this function name always catches me off guard as it's a _create and persist_ unlike the other functions on  `localEchoEventFactory`

Reply:
I agree it can be confusing, I will try to find a better name for it.

Reply:
I renamed it, see [efa356e](https://github.com/vector-im/element-android/pull/6486/commits/efa356e5ab91908044d49704ca2e181e12a773c5)